### ISSUE TITLE
disableVisualSelection doesn't work for headers

### ISSUE THREAD

**Issue by swistach** [handsontable/handsontable#6025]
### Description
`disableVisualSelection` doesn't work for headers. It doesn't matter if we set `'header'` or `true` - headers will be highlighted.

Related issue: #5869

### Steps to reproduce
1. Initialize Handsontable with the following configuration:
```
rowHeaders: true,
colHeaders: true,
disableVisualSelection: true,
```
2. Select column `A` or 2nd row by a header.

### Demo
https://jsfiddle.net/handsoncode/8ffpsqt6/

### Your environment
* Handsontable version: 2.0.0+
* Browser Name and version: any
* Operating System: any

**Comment by aninde** [comment]
This bug can be repricable with `8.0.0-beta2-rev19` https://jsfiddle.net/aninde/92wa1txc/

**Comment by pnowak** [comment]
This issue is still replicable using v8.0.0. Here's an updated demo - https://jsfiddle.net/xvq9yw8u/

**Comment by AMBudnik** [comment]
It works for me (header is not highlighted when we use `disableVisualSelection: 'header'`) on Chrome 87/Firefox 84/Edge (Windows 10) using v7.4.2 and newer https://jsfiddle.net/handsoncode/6m1j5kxd/

![image](https://user-images.githubusercontent.com/10757813/105353250-4c478900-5bef-11eb-95c9-317dfcec52db.png)

**Comment by wszymanski** [comment]
> It works for me (header is not highlighted when we use `disableVisualSelection: 'header'`) on Chrome 87/Firefox 84/Edge (Windows 10) using v7.4.2 and newer [jsfiddle.net/handsoncode/6m1j5kxd](https://jsfiddle.net/handsoncode/6m1j5kxd/)
> 
> ![image](https://user-images.githubusercontent.com/10757813/105353250-4c478900-5bef-11eb-95c9-317dfcec52db.png)
> 
> EDIT: I've just tested it with macOS Catalina and Chrome. Also works fine.

Issue is addressing problem for clicks on headers :)

**Comment by AMBudnik** [comment]
I'm closing this issue as solved and released under Handsontable v8.3.2
